### PR TITLE
ghc-7.10.1: cherry-pick Cabal bugfix for GHCJS.

### PR DESCRIPTION
**Pull request by cstrahan**
This pulls in the upstream Cabal bugfix for GHCJS, rather than using `sed` to hack around the problem.

I'm currently rebuilding GHC 7.10.1 and GHCJS to test this out.

### CODE REVIEW

**Comment by cstrahan** [comment]
I have no idea why Travis is still complaining about "attribute ‘isGhcjs’ missing".

**Comment by abbradar** [comment]
That was a `master` problem already fixed, rebase and check again. BTW, thanks for your GHCJS work!

File: pkgs/development/compilers/ghc/7.10.1.nix [hunk 1]
```diff
@@ -28,9 +35,14 @@ stdenv.mkDerivation rec {
 
   enableParallelBuilding = true;
 
-  # We patch Cabal for GHCJS. See: https://github.com/haskell/cabal/issues/2454
+  postPatch = ''
+    (
```
Comment (peti) [inline pkgs/development/compilers/ghc/7.10.1.nix]: Won't the use of a bracketed expression here mean that error codes during patching will be lost / ignored by the builder?

File: pkgs/development/compilers/ghc/7.10.1.nix [hunk 2]
```diff
@@ -28,9 +35,14 @@ stdenv.mkDerivation rec {
...
+      cd libraries/Cabal
+      cat ${cabalPatch} | patch -p1
```
Comment (peti) [inline pkgs/development/compilers/ghc/7.10.1.nix]: The use of `cat` here seems somewhat unnecessary. :grin:

**Comment by peti** [comment]
028779f fixes the evaluation error; there should be no need for 06a54bf2c49742117219c86d3324bbb5360671eb.

File: pkgs/development/compilers/ghc/7.10.1.nix (hunk 1 above)
Comment (cstrahan) [inline pkgs/development/compilers/ghc/7.10.1.nix]: > Won't the use of a bracketed expression here mean that error codes during patching will be lost / ignored by the builder?

Bracketed expressions appear to exit with the same code as the last executed expression contained therein, but I might as well use an alternative (`pushd`/`popd`, perhaps), just to be a little less error prone.

File: pkgs/development/compilers/ghc/7.10.1.nix (hunk 2 above)
Comment (cstrahan) [inline pkgs/development/compilers/ghc/7.10.1.nix]: > The use of cat here seems somewhat unnecessary. :grin:

Ah, if you mean that I should have just used `< ${cabalPatch}`, then I agree.

**Comment by cstrahan** [comment]
@abbradar Sure thing!

@peti I think my last push addresses your points. From my testing, everything appears to be working properly.

**Pull request merged**